components/DashboardPage/components/Pomodoro/Timer.tsx

```diff
@@ -0,0 +1,103 @@
+import moment from 'moment';
+import { useState, useEffect } from 'react';
+import { Stack, Typography, Button } from 'components';
+
+interface TimerProps {
+  round: number;
+  setRound: any;
+}
+
+export function Timer({ round, setRound }: TimerProps) {
+  let workTime = 25 * 60;
+  let breakTime = 5 * 60;
+  const [time, setTime] = useState<number>(workTime);
+  const [clock, setClock] = useState<boolean>(true);
+  const [countdown, setCountdown] =
+    useState<boolean>(false);
+  const [workState, setWorkState] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    if (time === 0) {
+      setWorkState((workState) => !workState);
```

Comment:
is it necessary to catch the object as an argument instead of just doing 
```
setWorkState(!workState);
```
Same comment for line **25**. 
Concerning line 25 actually, the compiler is being tricked that the `round` prop is never used because of that notation.

Reply:
I think it's best practice to do so when the new state is depending on the older state. Don't ask me why or how tho 👀😅 @YannCedric can answer that better

Reply:
Ouais, in this case not much difference.
But sometimes, inside an effect you might have an older reference to a variable like `workState`. So to avoid any possible issue, get the previous value directly from the function.